web-common/src/lib/time/ranges/iso-ranges.ts

```diff
@@ -0,0 +1,189 @@
+import { convertTimeRangePreset } from "@rilldata/web-common/lib/time/ranges/index";
+import { transformDate } from "@rilldata/web-common/lib/time/transforms";
+import {
+  Period,
+  RelativeTimeTransformation,
+  TimeOffsetType,
+  TimeRange,
+  TimeRangePreset,
+  TimeTruncationType,
+} from "@rilldata/web-common/lib/time/types";
+import { Duration, parse, toString, subtract } from "duration-fns";
+
+/**
+ * Converts an ISO duration to a time range.
+ * Pass in the anchor to specify when the range should be from.
+ * NOTE: This should only be used for default time range. UI presets have their own settings.
+ */
+export function isoDurationToTimeRange(
+  isoDuration: string,
+  anchor: Date,
+  zone = "Etc/UTC"
+) {
+  const startTime = transformDate(
+    anchor,
+    getStartTimeTransformations(isoDuration),
+    zone
+  );
+  const endTime = transformDate(
+    anchor,
+    getEndTimeTransformations(isoDuration),
+    zone
+  );
+  return {
+    startTime,
+    endTime,
+  };
+}
+
+export const ISODurationToTimeRangePreset: Record<
+  string,
+  keyof typeof TimeRangePreset
+> = {
+  PT6H: TimeRangePreset.LAST_SIX_HOURS,
+  PT24H: TimeRangePreset.LAST_24_HOURS,
+  P1D: TimeRangePreset.LAST_24_HOURS,
+  P7D: TimeRangePreset.LAST_7_DAYS,
+  P14D: TimeRangePreset.LAST_14_DAYS,
+  P4W: TimeRangePreset.LAST_4_WEEKS,
+  inf: TimeRangePreset.ALL_TIME,
+};
+export function isoDurationToFullTimeRange(
+  isoDuration: string,
+  start: Date,
+  end: Date,
+  zone = "Etc/UTC"
+): TimeRange {
+  if (!isoDuration) {
+    return convertTimeRangePreset(TimeRangePreset.ALL_TIME, start, end, zone);
+  }
+  if (isoDuration in ISODurationToTimeRangePreset) {
+    return convertTimeRangePreset(
+      ISODurationToTimeRangePreset[isoDuration],
+      start,
+      end,
+      zone
+    );
+  }
+
+  const { startTime, endTime } = isoDurationToTimeRange(isoDuration, end, zone);
+  return {
+    name: TimeRangePreset.DEFAULT,
+    start: startTime,
+    end: endTime,
+  };
+}
+
+export function humaniseISODuration(isoDuration: string): string {
```

Comment:
I suppose we can directly utilize  Luxon methods to convert an ISO string to human readable duration? https://moment.github.io/luxon/api-docs/index.html#durationtohuman